#!/usr/bin/env python
PACKAGE = "tj2_gameobjects"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("meas_std_val", double_t, 0, "meas_std_val", 1.0,  0.0, 10.0)
gen.add("u_std_vx", double_t, 0, "u_std_vx", 0.01,  0.0, 10.0)
gen.add("u_std_vy", double_t, 0, "u_std_vy", 0.01,  0.0, 10.0)
gen.add("u_std_vz", double_t, 0, "u_std_vz", 0.01,  0.0, 10.0)
gen.add("u_std_vt", double_t, 0, "u_std_vt", 0.01,  0.0, 10.0)
gen.add("initial_range_x", double_t, 0, "initial_range_x", 1.0,  0.0, 10.0)
gen.add("initial_range_y", double_t, 0, "initial_range_y", 1.0,  0.0, 10.0)
gen.add("initial_range_z", double_t, 0, "initial_range_z", 1.0,  0.0, 10.0)
gen.add("initial_range_vx", double_t, 0, "initial_range_vx", 0.25,  0.0, 10.0)
gen.add("initial_range_vy", double_t, 0, "initial_range_vy", 0.25,  0.0, 10.0)
gen.add("initial_range_vz", double_t, 0, "initial_range_vz", 0.25,  0.0, 10.0)
gen.add("num_particles", int_t, 0, "num_particles", 150,  0, 1000)
gen.add("stale_filter_time", double_t, 0, "stale_filter_time", 0.1,  0.0, 10.0)
gen.add("lower_bound_x", double_t, 0, "lower_bound_x", 10.0, -100.0, 100.0)
gen.add("upper_bound_x", double_t, 0, "upper_bound_x", 10.0, -100.0, 100.0)
gen.add("lower_bound_y", double_t, 0, "lower_bound_y", 10.0, -100.0, 100.0)
gen.add("upper_bound_y", double_t, 0, "upper_bound_y", 10.0, -100.0, 100.0)
gen.add("lower_bound_z", double_t, 0, "lower_bound_z", 10.0, -100.0, 100.0)
gen.add("upper_bound_z", double_t, 0, "upper_bound_z", 10.0, -100.0, 100.0)
gen.add("lower_bound_vx", double_t, 0, "lower_bound_vx", 10.0, -100.0, 100.0)
gen.add("upper_bound_vx", double_t, 0, "upper_bound_vx", 10.0, -100.0, 100.0)
gen.add("lower_bound_vy", double_t, 0, "lower_bound_vy", 10.0, -100.0, 100.0)
gen.add("upper_bound_vy", double_t, 0, "upper_bound_vy", 10.0, -100.0, 100.0)
gen.add("lower_bound_vz", double_t, 0, "lower_bound_vz", 10.0, -100.0, 100.0)
gen.add("upper_bound_vz", double_t, 0, "upper_bound_vz", 10.0, -100.0, 100.0)
gen.add("velocity_smooth_k", double_t, 0, "velocity_smooth_k", 0.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "tj2_gameobjects", "ParticleFilter"))
